### PR TITLE
HTTP.sys WS config+example updates

### LINKED ISSUE #10164: HttpSysException: 'Access is denied'
I downloaded the sample app and also tried creating a new one from scratch based on this article, but I can get either to run. It always fails with the subject error on 'Microsoft.AspNetCore.Server.HttpSys.HttpSysListener[0] Start'. How do I determine what access is being denied or what configuration I need to change/add?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5e347c54-6a6a-d685-4af7-39b023bf7f15
* Version Independent ID: 41aa77f5-9466-6580-5ec6-b1bf97bd0f01
* Content: [HTTP.sys web server implementation in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys?view=aspnetcore-2.2)
* Content Source: [aspnetcore/fundamentals/servers/httpsys.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/servers/httpsys.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #10164 

[Internal Review Topic (Configure Windows Server section)](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/httpsys?view=aspnetcore-2.0&branch=pr-en-us-10186#configure-windows-server)

@marqdouj, you won't be able to see the built internal topic ... it's only visible to staff. You can flip over to the diff to see the proposed updates. If you get a chance to look at this, let me know if you feel all of the bases are covered here.

TL;DR the list of little enhancements I'm going for here. However, you can go straight to the diff ... it's all there.

* References to "certificate hash" should be changed to "thumbprint" because it's immediately recognizable and reported that way in cert properties.
* Note installing the cert for the Local Machine in the Personal store.
* Note that the `appid` is informational and generated by the dev for the `add sslcert` command.
* For the example code and commands, use an IP example ... e.g., `options.UrlPrefixes.Add("https://10.0.0.4:443");`.
* In addition to opening required ports in the WS firewall, we may as well remind readers using Azure VM's to open the port(s) in their NSG.
* For reference, show the commands for `delete urlacl` and `delete sslcert`.
* Update the sample app to 2.2 (and let's drop the static assets while we're at it).
* Add note that for an SCD no installs are required on the server.
* Update the .NET Core download links.
* Re-organize a bit ... currently, server activities are split across the app steps. Consolidate the server steps and app steps.
* Show commands with placeholders and example entries.

### CODE REVIEW

**Comment by guardrex** [comment]
I'll add a note for this on the next commit ...

```
services.AddAuthentication(Microsoft.AspNetCore.Server.HttpSys.HttpSysDefaults.AuthenticationScheme);
```

https://stackoverflow.com/questions/45818095/asp-net-core-2-0-httpsys-windows-authentication-fails-with-authorize-attribute

File: aspnetcore/fundamentals/servers/httpsys.md [hunk 1]
```diff
@@ -142,48 +152,93 @@ HTTP.sys delegates to kernel mode authentication with the Kerberos authenticatio
...
+   Administrator privileges aren't required to run the app when binding to localhost using HTTP (not HTTPS) with a port number greater than 1024. For other configurations (for example, using a local IP address or binding to port 443), run the app with administrator privileges.
 
-   2. Create self-signed X.509 certificates, if required.
+   The app responds at the server's public IP address.
 
-      [!INCLUDE [How to make an X.509 cert](~/includes/make-x509-cert.md)]
+   A development certificate is used in this example. The page loads securely after bypassing the browser's untrusted certificate warning.
 
-4. Open firewall ports to allow traffic to reach HTTP.sys. Use *netsh.exe* or [PowerShell cmdlets](https://technet.microsoft.com/library/jj554906).
+   ![Browser window showing the app's Index page loaded](httpsys/_static/browser.png)
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/httpsys.md]: This image is confusing as the address bar gives one IP and the page prints another. If you're going to keep this then mention they don't match because you were using a cloud hosted VM where the public IP is different from the local one.

File: aspnetcore/fundamentals/servers/httpsys.md [hunk 2]
```diff
@@ -142,48 +152,93 @@ HTTP.sys delegates to kernel mode authentication with the Kerberos authenticatio
...
+   netsh http delete urlacl url=<URL>
+   ```
 
-   If URL prefixes aren't preregistered in Windows, run the app with administrator privileges. The only exception is when binding to localhost using HTTP (not HTTPS) with a port number greater than 1024. In that case, administrator privileges aren't required.
+1. Register X.509 certificates on the server.
+
+   Use the *netsh.exe* tool to register certificates for the app:
+
+   ```console
+   netsh http add sslcert ipport=<IP>:<PORT> certhash=<THUMBPRINT> appid="{<GUID>}"
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/fundamentals/servers/httpsys.md]: Great idea ... let's note to the dev a good spot. Soooooooo ...... spit ball'in here ... what we could do is tell them to:

* If using VS, enter it in project's properties > **Package** > **Tags**.
* If not using VS, provide it in the project file in a `<PackageTags>` element:

  ```xml
  <PropertyGroup>
    <PackageTags>9412ee86-c21b-4eb8-bd89-f650fbf44931</PackageTags>
  </PropertyGroup>
  ```
It avoids the ugly VS "Edit project file" experience for folks who love property page app configuration. MSBuild :heart: this approach.

File: aspnetcore/fundamentals/servers/httpsys.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/servers/httpsys.md]: Ah, yes, I forgot/missed adding the public IP to the line leading into this ...

> The app responds at the server's public IP address.

I'll enhance this on the next commit.

File: aspnetcore/fundamentals/servers/httpsys.md [hunk 3]
```diff
@@ -128,12 +128,27 @@ HTTP.sys delegates to kernel mode authentication with the Kerberos authenticatio
 
 ### Configure Windows Server
 
+In the example app and server configuration in this section:
+
+* The server's local IP address is `10.0.0.4`.
+* The server's public IP address is `104.214.79.47`.
+
+1. Determine the ports to open for the app and use Windows Firewall or [PowerShell cmdlets](https://technet.microsoft.com/library/jj554906) to open firewall ports to allow traffic to reach HTTP.sys. When deploying to an Azure VM, open the ports in the [Network Secruity Group](/azure/virtual-network/security-overview). In the following commands and app configuration, port 443 is used.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/servers/httpsys.md]: Secruity typo

File: aspnetcore/fundamentals/servers/httpsys.md [hunk 4]
```diff
@@ -128,12 +128,27 @@ HTTP.sys delegates to kernel mode authentication with the Kerberos authenticatio
...
+   * **.NET Core** &ndash; If the app requires .NET Core, obtain and run the **.NET Core Runtime** installer from [.NET Core Downloads](https://dotnet.microsoft.com/download). Don't install the full SDK on the server.
+   * **.NET Framework** &ndash; If the app requires .NET Framework, see the [.NET Framework installation guide](/dotnet/framework/install/). Install the required .NET Framework. The installer for the latest .NET Framework is available from from the [.NET Core Downloads](https://dotnet.microsoft.com/download) page.
+
+   If the app is a [self-contained deployment](/dotnet/core/deploying/#framework-dependent-deployments-scd), the app includes the runtime in its deployment. No framework installation is required on the server.
 
-2. Configure URLs and ports for the app.
+1. Configure URLs and ports in the app.
 
    By default, ASP.NET Core binds to `http://localhost:5000`. To configure URL prefixes and ports, options include using:
```
Comment (tdykstra) [inline aspnetcore/fundamentals/servers/httpsys.md]: This might sound better if we drop the word "using" or change "option include using:" with something like "choose one of the following options:"

File: aspnetcore/fundamentals/servers/httpsys.md [hunk 5]
```diff
@@ -128,12 +128,27 @@ HTTP.sys delegates to kernel mode authentication with the Kerberos authenticatio
 
 ### Configure Windows Server
 
+In the example app and server configuration in this section:
```
Comment (tdykstra) [inline aspnetcore/fundamentals/servers/httpsys.md]: This sentence and the following two bullets seem unnecessary since each place that uses these addresses explicitly explains them.
